src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py

```diff
@@ -247,28 +337,285 @@ def test_keyvault_mgmt(self, resource_group):
+        # download SD
+        self.cmd('az keyvault security-domain download --hsm-name {hsm_name} --security-domain-file "{sdfile}" '
+                 '--sd-quorum 2 --sd-wrapping-keys "{pem_dir}/sd1.cer" "{pem_dir}/sd2.cer" "{pem_dir}/sd3.cer"')
+
+        # delete the HSM
+        self.cmd('az group lock delete -g {rg} -n {rg_lock}')
+        self.cmd('az keyvault delete --hsm-name {hsm_name}')
+        self.cmd('az group lock create -g {rg} -n {rg_lock} -t CanNotDelete')
+
+        # create a new HSM
+        self.cmd('az keyvault create --hsm-name {next_hsm_name} -l {loc} -g {rg} --administrators {init_admin} '
+                 '--retention-days 7 --no-wait')
+
+        # wait until the HSM is ready for recovery
+        self.cmd('az keyvault wait-hsm -n {next_hsm_name} '
+                 '--custom "statusMessage!=\'Resource creation in progress. Allocating hardware...\'"')
+
+        # download the exchange key
+        self.cmd('az keyvault security-domain init-recovery --hsm-name {next_hsm_name} '
+                 '--sd-exchange-key "{exchange_key}"')
+
+        # upload the blob
+        self.cmd('az keyvault security-domain upload --hsm-name {next_hsm_name} --sd-file "{sdfile}" '
+                 '--sd-exchange-key "{exchange_key}" '
+                 '--sd-wrapping-keys "{pem_dir}/sd1.key" "{pem_dir}/sd2.key"')
+
+        # restore the key
+        self.cmd('az keyvault key restore --hsm-name {next_hsm_name} -f "{key_backup}"')
+
+
+class KeyVaultHSMFullBackupRestoreScenarioTest(ScenarioTest):
+    @pytest.mark.serial()
+    @ResourceGroupPreparer(name_prefix='cli_test_keyvault_hsm_full_backup')
+    @AllowLargeResponse()
+    def test_keyvault_hsm_full_backup_restore(self):
+        self.kwargs.update({
+            'hsm_url': ACTIVE_HSM_URL,
+            'hsm_name': ACTIVE_HSM_NAME,
+            'storage_account': self.create_random_name('clitesthsmsa', 24),
+            'blob': self.create_random_name('clitesthsmblob', 24),
+            'sas_start': (datetime.utcnow() - timedelta(minutes=30)).strftime('%Y-%m-%dT%H:%M:%SZ'),
+            'sas_expiry': (datetime.utcnow() + timedelta(minutes=30)).strftime('%Y-%m-%dT%H:%M:%SZ')
+        })
+        self.cmd('az storage account create -n {storage_account} -g {rg}')
+        self.cmd('az storage container create -n {blob} --account-name {storage_account} -g {rg}')
+
+        self.kwargs['sas'] = '?' + self.cmd('az storage account generate-sas --start {sas_start} --expiry {sas_expiry} '
+                                            '--https-only '
+                                            '--permissions rwdlacu --resource-types sco --services b '
+                                            '--account-name {storage_account}').get_output_in_json().replace('%3A', ':')
+
+        self.cmd('az keyvault backup start --id {hsm_url} --blob-container-name {blob} '
+                 '--storage-account-name {storage_account} '
+                 '--storage-container-SAS-token "{sas}"',
```

Comment:
typically sas related command is hard to playback because there is random signature. Does it work in your scenrio?

Reply:
@Juliehzl This works. I generate a new SAS every time.